### ISSUE TITLE
Knex.js and enclose

### ISSUE THREAD

**Issue by arnaudchenyensu** [tgriesser/knex#1450]
When using [enclose](https://github.com/igorklopov/enclose), for some reasons I needed to install the package `pg-query-stream` (see this [issue](https://github.com/igorklopov/enclose/issues/96)).

Not sure if this is a specific issue to `enclose` or a missing dependency for `knex.js`.

**Comment by jurko-gospodnetic** [comment]
I think this is actually documented here: http://knexjs.org#Interfaces-Streams

Not sure why it is not listed as an explicit dependency. Possibly so that people not using Postgres would not need to install it?

**Comment by rhys-vdw** [comment]
> Possibly so that people not using Postgres would not need to install it?

Yep. Same deal with the various db libraries (`pg`, `mysql` etc). These could (and should?) be listed under `optionalDependencies` to lock down their versions.

**Comment by mercmobily** [comment]
Closing this in favour of #6230
